### ISSUE TITLE
[ENH] Rethink the mask parameter in similarity search

### ISSUE THREAD

**Issue by baraline** [aeon-toolkit/aeon#2020]
### Describe the issue

Currently, a boolean mask is created to indicate which part of the input time series dataset is to be ignored from the computations if the query used in the search is included in the input time series dataset. 

The issue is that the computations linked to these areas are still made, they are only removed from the result array before returning it.

### Suggest a potential alternative/fix

Rethink the mask parameter to be able to skip computation that won't be used instead of ignoring them later on.

**Comment by vedpawar2254** [comment]
@aeon-actions-bot assign @vedpawar2254

**Comment by MatthewMiddlehurst** [comment]
@vedpawar2254 please don't take on multiple issues at once unless the others are close to completion.

**Comment by vedpawar2254** [comment]
Hi, @MatthewMiddlehurst I completed the work and sebastian has reviewed it, I'll update the PR tonight

**Comment by MatthewMiddlehurst** [comment]
@vedpawar2254 you assigned https://github.com/aeon-toolkit/aeon/issues/2081

**Comment by vedpawar2254** [comment]
Yes @MatthewMiddlehurst , I am working on that rn but it's kind of getting the better of so I though I'll try something relatively easier so I can start on that one again with a fresh mind, i'll unassign myself if you want
